src/controls/adaptiveCardDesignerHost/AdaptiveCardDesigner.tsx

```diff
@@ -0,0 +1,229 @@
+import * as monacoLoader from '@monaco-editor/loader';
+import { Versions } from 'adaptivecards';
+import { BindingPreviewMode, CardDesigner, FieldDefinition, GlobalSettings, HostContainer, ToolbarChoicePicker } from 'adaptivecards-designer';
+import * as markdownit from "markdown-it";
+import { Spinner, SpinnerSize } from 'office-ui-fabric-react/lib/Spinner';
+import * as React from 'react';
+import { useEffect, useRef, useState } from 'react';
+import { injectContextProperty } from '../adaptiveCardHost/AdaptiveCardHost.Utilities';
+import { fluentUIDefaultTheme } from '../../common/fluentUIThemes/FluentUIDefaultTheme';
+import { addToolbarButton, addToolbarChoicePicker, addToolboxSnippet, convertNullToEmptyString, hideToolbarElement } from './AdaptiveCardDesigner.Helpers';
+import { applyAdaptiveCardDesignerStyles } from './AdaptiveCardDesigner.Styles';
+import { AdaptiveCardHostContainer, AdaptiveCardHostContainerType } from './fluentUI/AdaptiveCardHostContainer';
+import { initializeDesignerPeers } from './fluentUI/peers/DesignerPeers';
+import { IAdaptiveCardDesignerHostProps } from './IAdaptiveCardDesignerProps';
+
+export const EmptyCard = {
+    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+    "type": "AdaptiveCard",
+    "version": "1.5"
+};
+
+export const AdaptiveCardDesigner = (props: IAdaptiveCardDesignerHostProps) => {
```

Comment:
As you're actively working with the `props` - maybe makes sense to unbox the object?

Reply:
If it is mandatory I do, otherwise I always prefer to use parameters directly from "props."

Reply:
no, it's just a nit. Leave it as is if you prefer so.